devito/ops/transformer.py

```diff
@@ -244,13 +267,23 @@ def create_ops_arg(p, name_to_ops_dat, par_to_ops_stencil):
             namespace['ops_read']
         )
     else:
-        return namespace['ops_arg_dat'](
-            name_to_ops_dat[p.name],
-            1,
-            par_to_ops_stencil[p],
-            Literal('"%s"' % dtype_to_cstr(p.dtype)),
-            namespace['ops_read'] if p.read_only else namespace['ops_write']
-        )
+        accessible_info = accessible_origin[p.name]
+
+        if accessible_info.time:
```

Comment:
This if-the-else takes 15 lines, but the difference in the `return` statement is only in one argument... it's a real pity... can we not improve this? Ideally I'd have

```
return namespace['ops_arg_dat'](....., 1, par_to_ops_stencil[p], Literal(...) ...)
```

Reply:
Soo.. I end it up needing to add a new if/else...
But i followed your advise and tried to shorten it. 

But the previous if/else that existed I didn't manage to improve, because it was getting messy and confused IMO.